File: Sorting/Sorting03.py
def solution(citations):
    i = 0
    j = 0
    hMax = 0
    moreThanHCount = 0

    while (i < len(citations)):
        while (j < len(citations)):
            if (hMax <= citations[j]):
                moreThanHCount = moreThanHCount + 1
            j = j + 1
        if (hMax <= moreThanHCount):
            hMax = hMax + 1
            j = 0
            i = i + 1
            moreThanHCount = 0
        else:
            return hMax - 1
    return hMax

File: Sorting/Sorting03.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<int> citations) {
	int answer = 0;
	sort(citations.begin(), citations.end());
	answer=citations.back();
    
    while(answer!=0){
        int count=0;
        for(int i=citations.size()-1;citations[i]>=answer;i--){
            count++;
        }
        if(count>=answer)
            break;
        answer--;
    }
	return answer;
}